src/provider/provider_tracking.c

```diff
@@ -267,6 +267,61 @@ static umf_result_t trackingAllocationSplit(void *hProvider, void *ptr,
     return ret;
 }
 
+// shrink (or remove) an entry in the tracker and return the totalSize of the original entry
+umf_result_t trackerShrinkEntry(void *hProvider, void *ptr, size_t shrinkSize,
+                                size_t *totalSize) {
+    umf_result_t ret = UMF_RESULT_ERROR_UNKNOWN;
+    umf_tracking_memory_provider_t *provider =
+        (umf_tracking_memory_provider_t *)hProvider;
+
+    int r = utils_mutex_lock(&provider->hTracker->splitMergeMutex);
+    if (r) {
+        return UMF_RESULT_ERROR_UNKNOWN;
+    }
+
+    tracker_alloc_info_t *value = (tracker_alloc_info_t *)critnib_get(
+        provider->hTracker->alloc_segments_map, (uintptr_t)ptr);
+    if (!value) {
+        LOG_ERR("region for shrinking is not found in the tracker");
+        ret = UMF_RESULT_ERROR_INVALID_ARGUMENT;
+        goto err;
+    }
+    if (shrinkSize > value->size) {
+        LOG_ERR("requested size %zu to shrink exceeds the tracked size %zu",
+                shrinkSize, value->size);
+        ret = UMF_RESULT_ERROR_INVALID_ARGUMENT;
+        goto err;
+    }
+
+    if (shrinkSize < value->size) {
+        void *highPtr = (void *)(((uintptr_t)ptr) + shrinkSize);
+        size_t secondSize = value->size - shrinkSize;
```

Comment:
The `shrinkSize` parameter seems to be a size that the tracker entry should be reduced by, but in the `fixed_initialize` the size provided is `in_params->size` which is the target size for the fixed provider I think.

Reply:
Yes, and this is correct - we have to remove from the tracker the memory range that will be used by Fixed-size memory provider.

Reply:
Oh I see, we shrink from the low address side and bump up the ptr.